Add xcpretty_output option to spm action

taisukeh opened the pull request:
<!-- Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes by putting an x in the [ ] (don't: [x ], [ x], do: [x]) -->

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->
<!-- Please describe in detail how you tested your changes. -->

I use `spm`( swift package manager ) action to run tests of a SwiftPM library. But the output is difficult to see.

![image](https://user-images.githubusercontent.com/282077/39621948-7af8e3b2-4fcb-11e8-9175-0526ffd62480.png)

### Description
<!-- Describe your changes in detail -->

The xcpretty works well to summarize the `swift test` output. So I add `xcpretty_output` option to the `spm` action to enable xcpretty.

![image](https://user-images.githubusercontent.com/282077/39622027-b9b43656-4fcb-11e8-9e10-302585164518.png)

taisukeh committed:
Add xcpretty_output option to spm action

joshdholtz commented on fastlane_core/lib/fastlane_core/command_executor.rb:
```diff
@@ -47,6 +47,8 @@ def execute(command: nil, print_all: false, print_command: true, error: nil, pre
           UI.command_output(loading)
         end
 
+        return command unless Helper.sh_enabled? # only for the tests
```
I know the `sh` action works like this but I would like to keep any `Helper.sh_enabled?` outside of main code as much as possible (and since `FastlaneCommand::Executor` isn't `sh`) 😇 

So for now I think the best solution is here to probably do something like this in your tests...
```rb
before do
  # Force FastlaneCore::CommandExecutor.execute to return command
  allow(FastlaneCore::CommandExecutor).to receive(:execute).and_wrap_original do |m, *args|
    args[0][:command]
  end
end
```

taisukeh replied:
@joshdholtz Fixed. Your code worked perfectly! Thank you for the good solution 🙇

joshdholtz reviewed (changes requested):
Small nit pick 😊

taisukeh committed:
Mock FastlaneCore::CommandExecutor.execute

- Revert `command_executor.rb`
- Mock FastlaneCore::CommandExecutor.execute in `spm_spec.rb`

joshdholtz reviewed (approved):
I don't have anything to run this with myself but the tests look good and pass so let's merge this one 👍

joshdholtz commented:
@taisukeh Thanks for the contribution!
